packages/expo-clipboard/android/src/main/java/expo/modules/clipboard/ClipboardModule.kt

```diff
@@ -5,36 +5,91 @@ import android.content.ClipData
 import android.content.ClipboardManager
 import android.util.Log
 import androidx.core.os.bundleOf
+import expo.modules.core.errors.ModuleDestroyedException
 import expo.modules.core.utilities.ifNull
+import expo.modules.kotlin.Promise
 import expo.modules.kotlin.exception.CodedException
 
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import java.io.File
 
 private const val moduleName = "ExpoClipboard"
-const val clipboardChangedEventName = "onClipboardChanged"
 private val TAG = ClipboardModule::class.java.simpleName
 
-const val ERR_CLIPBOARD_UNAVAILABLE = "ERR_CLIPBOARD_UNAVAILABLE"
-
-class ClipboardUnavailableException :
-  CodedException(ERR_CLIPBOARD_UNAVAILABLE, "'CLIPBOARD_SERVICE' is unavailable on this device", null)
+// this must match the one from `res/xml/clipboard_provider_paths.xml`
+const val clipboardDirectoryName = ".clipboard"
+const val clipboardChangedEventName = "onClipboardChanged"
 
 class ClipboardModule : Module() {
   override fun definition() = ModuleDefinition {
     name(moduleName)
 
+    // region Strings
     function("getStringAsync") {
-      val clip = clipboardManager.primaryClip?.takeIf { it.itemCount >= 1 }
-      clip?.getItemAt(0)?.text ?: ""
+      clipboardManager.firstItem?.text ?: ""
     }
 
     function("setStringAsync") { content: String ->
       val clip = ClipData.newPlainText(null, content)
       clipboardManager.setPrimaryClip(clip)
       return@function true
     }
+    // endregion
+
+    // region Images
+    function("getImageAsync") { options: GetImageOptions, promise: Promise ->
+      val imageUri = clipboardManager
+        .takeIf { clipboardHasItemWithType("image/*") }
+        ?.firstItem
+        ?.uri
+        ?: return@function null
+
+      val exceptionHandler = CoroutineExceptionHandler { _, err ->
+        err.printStackTrace()
```

Comment:
I added printing stack trace in case somebody having issue is able to use logcat. Sweet modules API handles "decorated" exceptions, but I'm not sure if the root-cause-stacktrace is logged anywhere.
I don't have any strong opinion here, I'm open to any discussions/suggestions

Reply:
cc @tsapeta, @lukmccall

Reply:
I don't have a strong opinion, however, this seems to be a nice addition that doesn't cost anything ;)

Reply:
I've created [ENG-4268](https://linear.app/expo/issue/ENG-4268/improve-expo-modules-core-explore-auto-logging-upon-promise-rejection) to keep track of this one.